xclim/indices/_conversion.py

```diff
@@ -895,6 +903,59 @@ def rain_approximation(
     return prra
 
 
+def snow_depth_approximation(
+    snw: xr.DataArray,
+    snr: xr.DataArray | None = None,
+) -> xr.DataArray:
+    """Approximation of snow density from snow water.
+
+    Parameters
+    ----------
+    snw : xr.DataArray
+        Snow water (kg/m^2).
+        If snow thickness (`swe`: "m") is provided, will be converted to `snw` before calculating.
+    snr : xr.DataArray, optional
+        Snow density (kg/m^3).
```

Comment:
```suggestion
    """Approximation of snow depth from snow amount and density.

    Parameters
    ----------
    snw : xr.DataArray
        Snow amount [kg/m^2].
        If snow water equivalent (`swe` [m]) is provided instead, will be converted to `snw` before calculating.
    snr : xr.DataArray, optional
        Snow density [kg/m^3].
```
However, I'm not sure swe would pass the unit check? This is something I wanted to work on this year maybe : that the check fails if conversion is impossible, instead of through an explicit dimension check. 
For the moment, I don't think we need to support swe as the conversion from swe to snw is litterally a `convert_units_to` call, as you in fact did.

Reply:
Works for me!